src/odemis/gui/cont/acquisition/cryo_z_localization.py

```diff
@@ -68,51 +74,274 @@ def __init__(self, tab_data, panel, tab):
+            self._target_z_va_connector.disconnect()
+
+        # Refresh the viewports to show the selected target
+        for vp in self._viewports:
+            vp.canvas.update_drawing()
+
+        # Update the text display of the current target
+        if target is None:
+            self._enable_target_ctrls(False)
+            return
+        elif target and (target.type.value == TargetType.Fiducial) and not self.current_target_coordinate_subscription:
+            self._enable_target_ctrls(True)
+            self._target_z_va_connector = VigilantAttributeConnector(target.coordinates,
+                                                                      self._panel.ctrl_target_z,
+                                                                      events=wx.EVT_TEXT_ENTER,
+                                                                      ctrl_2_va=self._on_ctrl_target_z_change,
+                                                                      va_2_ctrl=self._on_target_focus_pos)
+
+            self._tab_data.main.currentTarget.value.coordinates.unsubscribe(self._on_current_coordinates_changes)
+            self._tab_data.main.currentTarget.value.coordinates.subscribe(self._on_current_coordinates_changes,
+                                                                                init=True)
+
+            # subscribe only once
+            self.current_target_coordinate_subscription = True
+
+    @call_in_wx_main
+    def _on_current_coordinates_changes(self, coordinates: ListVA) -> None:
+        """
+        Update the coordinates of the current target in the grid and update the correlation result.
+        :param coordinates: the coordinates of the current target
+        """
+        self.current_target_coordinate_subscription = False
+
+        self._update_target_cmb_list()
+
+        # Update the correlation target
+        self.correlation_target = update_feature_correlation_target(self.correlation_target, self._tab_data)
+
+    def _on_delete_target(self, evt) -> None:
+        """
+        Deletes the currently selected target.
+        """
+        if not self._tab_data.main.currentTarget.value:
+            return
+
+        for target in self._tab_data.main.targets.value:
+            if target.name.value == self._tab_data.main.currentTarget.value.name.value:
+                logging.debug(f"Deleting target: {target.name.value}")
+                self._tab_data.main.targets.value.remove(target)
+                self._tab_data.main.currentTarget.value = None
+                break
+
+        self._update_target_cmb_list()
+        # Update the correlation target
+        self.correlation_target = update_feature_correlation_target(self.correlation_target, self._tab_data)
+        # Deletes the target from each viewport
+        for vp in self._viewports:
+            vp.canvas.update_drawing()
 
-    def _cmb_stig_angle_get(self):
+    def _cmb_stig_angle_get(self) -> float:
```

Comment:
The typing is incorrect. The function can return None -> `Optional[float]` (and change the `return` to `return None` to make that clearer).